src/types.rs

```diff
@@ -1653,46 +2190,222 @@ impl TRS {
+    ///
+    /// See [`Deterministic System`] for more information.
+    ///
+    /// [`Deterministic System`]: http://en.wikipedia.org/wiki/Deterministic_system
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # extern crate rand;
+    /// # extern crate term_rewriting;
+    /// # fn main(){
+    /// # use term_rewriting::{Signature, Rule, parse_rule, TRS, parse_trs};
+    /// # use rand::{thread_rng,Rng};
+    /// let mut sig = Signature::default();
+    /// 
+    /// let mut t = parse_trs(&mut sig, 
+    /// "A = B | C;
+    /// D = E;").expect("parse of A = B | C; D = E");
+    /// let mut r = rand::thread_rng();
+    ///
+    /// assert!(!t.is_deterministic());
+    /// t.make_deterministic(& mut r);
+    /// assert!(t.is_deterministic());
+    /// # }
+    /// ```
     pub fn is_deterministic(&self) -> bool {
         self.is_deterministic
     }
-    /// The number of rules in the TRS.
+    /// The number of [`Rule`]s in the `TRS`.
+    /// 
+    /// [`Rule`]: struct.Rule.html
+    ///
+    /// # Examples
+    /// ```
+    /// # use term_rewriting::{Signature, TRS, parse_trs};
+    /// let mut sig = Signature::default();
+    ///
+    /// let t = parse_trs(&mut sig,
+    /// "A = B;
+    /// C = D | E;
+    /// F(x_) = G;").expect("parse of A = B; C = D | E; F(x_) = G;");
+    /// 
+    /// assert_eq!(t.len(), 3);
+    /// ```
     pub fn len(&self) -> usize {
         self.rules.len()
     }
-    /// Are there any rules in the TRS?
+    /// Are there any [`Rule`]s in the `TRS`?
+    /// 
+    /// [`Rule`]: struct.Rule.html
+    ///
+    /// # Examples
+    /// ```
+    /// # use term_rewriting::{Signature, TRS, parse_trs};
+    /// let mut sig = Signature::default();
+    ///
+    /// let t = parse_trs(&mut sig,
+    /// "A = B;
+    /// C = D | E;
+    /// F(x_) = G;").expect("parse of A = B; C = D | E; F(x_) = G;");
+    /// 
+    /// assert!(!t.is_empty());
+    ///
+    /// let t = parse_trs(&mut sig, "").expect("parse of blank string");
+    /// 
+    /// assert!(t.is_empty());
+    /// ```
     pub fn is_empty(&self) -> bool {
         self.rules.is_empty()
     }
-    /// Return the number of total number of subterms across all rules.
+    /// Return the number of total number of subterms across all [`Rule`]s in the `TRS`.
+    ///
+    /// See [`Term`] for more information.
+    ///
+    /// [`Term`]: struct.Term.html
+    /// [`Rule`]: struct.Rule.html
+    /// 
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, TRS, parse_trs};
+    /// let mut sig = Signature::default();
+    ///
+    /// let t = parse_trs(&mut sig,
+    /// "A = B;
+    /// C = D | E;
+    /// F(x_) = G;").expect("parse of A = B; C = D | E; F(x_) = G;");
+    /// 
+    /// assert_eq!(t.size(), 8);
+    /// 
+    /// let t = parse_trs(&mut sig, 
+    /// "A(x_) = B;
+    /// C = D;").expect("parse of A(x_) = B; C = D;");
+    ///
+    /// assert_eq!(t.size(), 5);
```

Comment:
I don't think we need this second example (the first one is more complex and this second one doesn't seem to highlight anything new about the function), so unless you see any specific need for it, please delete it.